gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/op/AbstractEvalOpProcessor.java

```diff
@@ -222,9 +223,33 @@ protected AbstractEvalOpProcessor(final boolean manageTransactions) {
      *                                script evaluation.
      * @param bindingsSupplier A function that returns the {@link Bindings} to provide to the
      *                         {@link GremlinExecutor#eval} method.
+     * @see #evalOpInternal(ResponseHandlerContext, Supplier, BindingSupplier)
      */
     protected void evalOpInternal(final Context context, final Supplier<GremlinExecutor> gremlinExecutorSupplier,
```

Comment:
it was good that you kept the old signature as other people might depend on this. i was thinking we might deprecate, this version of the method, but perhaps that is not necessary or perhaps even undesired. I guess, the question is whether or not the new method would ever be called directly or would it always be called through this method? Perhaps the other method should be `private`?

Reply:
Yeah, I was thinking about that but could not really reach any decision myself, so I kept it more accessible. In current code no other classes need to call the new method, so I'm ok to make it `private`.

However, if we keep the new `evalOpInternal` method `private` by the same logic it might be reasonable to keep the new `makeFrame` method `private` as well. If other classes ever need to use it directly it would be under similar conditions as calling the new `evalOpInternal` method... and right now other classes do not need to do that.

What do you think?

Reply:
well - i guess, keep it open as `protected` - i guess if someone is writing one of these they will need to know how all of it works  i guess. perhaps this is where other PRs comes in for tp33 and master. maybe we deprecate in 3.2.x and then remove them for 3.4.0.